src/factories/calorimetry/CalorimeterClusterRecoCoG_factory.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023, Wouter Deconinck
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+#pragma once
+
+#include "algorithms/calorimetry/CalorimeterClusterRecoCoG.h"
+#include "services/geometry/dd4hep/DD4hep_service.h"
+#include "extensions/jana/JOmniFactory.h"
+
+
+namespace eicrecon {
+
+class CalorimeterClusterRecoCoG_factory : public JOmniFactory<CalorimeterClusterRecoCoG_factory, CalorimeterClusterRecoCoGConfig> {
+
+    using AlgoT = eicrecon::CalorimeterClusterRecoCoG;
+    std::unique_ptr<AlgoT> m_algo;
+
+    PodioInput<edm4eic::ProtoCluster> m_proto_input {this};
+    PodioInput<edm4hep::SimCalorimeterHit> m_mchits_input {this};
+
+    PodioOutput<edm4eic::Cluster> m_cluster_output {this};
+    PodioOutput<edm4eic::MCRecoClusterParticleAssociation> m_assoc_output {this};
+
+    ParameterRef<std::string> m_energyWeight {this, "energyWeight", config().energyWeight};
+    ParameterRef<double> m_samplingFraction {this, "samplingFraction", config().sampFrac};
+    ParameterRef<double> m_logWeightBase {this, "logWeightBase", config().logWeightBase};
+    ParameterRef<bool> m_enableEtaBounds {this, "enableEtaBounds", config().enableEtaBounds};
+
+    Service<DD4hep_service> m_geoSvc {this};
+    // Resource<DD4hep_service, const dd4hep::Detector*> m_detector {this, [](std::shared_ptr<DD4hep_service> s, int64_t run_nr){ return s->detector(); }}
+
+public:
+    void Configure() {
+        m_algo = std::make_unique<AlgoT>(GetPrefix());
+        m_algo->applyConfig(config());
+        m_algo->init(m_geoSvc().detector(), logger());
+    }
+
+    void ChangeRun(int64_t run_number) {
+    }
+
+    void Process(int64_t run_number, uint64_t event_number) {
+        m_cluster_output() = std::move(std::make_unique<PodioTypeMap<edm4eic::Cluster>::collection_t>());
```

Comment:
I feel like creating the (initially empty) output collections should be responsibility of JOmnifactory itself, if the Algorithm isn't going to do it. That way we can be confident we will always have empty collections as opposed to missing collections.

Reply:
Good point, I'll see if I can std::enable_if this in JOmniFactory based on some type in the algorithm.

Reply:
This turns out to be harder than I thought, since the JOmniFactory doesn't know anything about the algorithm.

Reply:
Addressed in [a42ba63](https://github.com/eic/EICrecon/pull/865/commits/a42ba6369499c754f3907ccaf89f326e1bef78e7). Not entirely happy with it, but good enough. Reasons:
- requires OutputBase to have additional public interface Reset(),
- Reset() is only used by PodioOutput,
- it requires the factories to expose AlgoT publicly, and we end up with `AlgoT::AlgoT::` as the type...